graphql_compiler/interpreter/typedefs.py

```diff
@@ -272,5 +272,103 @@ def can_coerce_to_type(
         neighbor_hints: Optional[Collection[Tuple[EdgeInfo, NeighborHint]]] = None,
         **hints: Any,
     ) -> Iterable[Tuple[DataContext[DataToken], bool]]:
-        """Determine if each of an iterable of input DataTokens can be coerced to another type."""
-        # TODO(predrag): Add more docs in an upcoming PR.
+        """Determine if each of an iterable of input DataTokens can be coerced to another type.
+
+        Consider a query like the following:
+            {
+                Foo {
+                    out_Foo_Bar {
+                        ... on BarImpl {
+                            < ... some fields here ... >
+                        }
+                    }
+                }
+            }
+
+        Assume that this query is written against a schema that contains the following definitions:
+            type Foo {
+                < ... some fields here ... >
+                out_Foo_Bar: [Bar]
+            }
+
+            interface Bar {
+                < ... some fields here ... >
+            }
+
+            type BarImpl implements Bar {
+                < ... some fields here ... >
+            }
+
+        When resolving the out_Foo_Bar edge in the query using project_neighbors(), the interpreter
+        receives DataTokens that (per the schema) are instances of the Bar interface type. However,
+        the query's subsequent "... on BarImpl" type coercion clause requires that the interpreter
+        discard any neighboring vertices that are not instances of BarImpl, a subtype of
+        the Bar interface type.
+
+        The interpreter uses can_coerce_to_type() for this purpose: it calls this function with
+        an iterable of DataContexts, with current_type_name set to "Bar" (the schema-implied type
+        of the query's scope) and with coerce_to_type_name set to "BarImpl" as the type to which
+        coercion is being attempted. For each DataContext in the input iterable, this function
+        yields a tuple containing the context itself together with a bool set to True if
```

Comment:
```suggestion
        yields a tuple containing the context itself with a bool set to True if
```

Reply:
I initially mis-parsed this as "context itself with a bool" being one clause. How about this instead:
`yields a tuple containing the context itself and a bool set to True ...`

Reply:
👍  yep, i like that better!